manifest/protobuf.go

```diff
@@ -137,23 +200,61 @@ func loadDescriptorSets(pkg *pbsubstreams.Package, manif *Manifest) ([]*desc.Fil
 			"https://buf.build",
 		)
 
-		request := connect.NewRequest(&reflectv1beta1.GetFileDescriptorSetRequest{
-			Module:  descriptor.Module,
-			Symbols: descriptor.Symbols,
-			Version: descriptor.Version,
-		})
+		var fileDescriptorSetResponse *descriptorpb.FileDescriptorSet
 
-		authToken := os.Getenv("BUFBUILD_AUTH_TOKEN")
-		if authToken != "" {
-			request.Header().Set("Authorization", "Bearer "+authToken)
-		}
+		// Don't use cache for mutable versions or empty versions
+		if !cache.shouldCache(descriptor.Version) {
```

Comment:
I think the inverse flow would read better and reduce the duplication you have now between the two if branches.

```
if cache.isDeterministicVersion(descriptor.Version) {
  desc, err := cache.load(...)
  if err == nil {
    return desc
  } else {
    // Log warning about cache retrieval failure, I'm even fine if it's just logged in DEBUG
  }
}

// Fetch & save from here as either it was not cached or some error happened.